### PR TITLE
feat(persons): Add formation patterns for Persons (Person ships)

### PR DESCRIPTION
**Pull request by petervdmeer**
This PR adds a feature for person ships as discussed in PR #10374

## Summary
This PR adds support for formations around person-ships.

NPC ships typically seem to be too busy planning their next action (landing or jumping) which results in fleets immediately going for landing or jumping, and as a result having the formations not appearing. Fixing those is part of other PRs, this PR is only for adding formations for person-ships.

## Screenshots
See #10374 for an example of a person-ship with a formation around it.

## Usage examples
```
person "Some Person"
	formation "Some Formation"
```

## Testing Done
I've seen a formation briefly appear for ships in #10378 (when they were not busy jumping or landing).

## Save File
N/A. But can be tested in combination with PR #10374.

## Wiki Update
Let's update the wiki when the other formation issues (of NPCs being too busy to form formations) are also fixed. We don't want people to try them today and get stuck.

## Performance Impact
None expected; formations are already supported on ships, it's just that they now also can be loaded on person ships.

### CODE REVIEW

File: source/Person.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include "Person.h"
 
 #include "DataNode.h"
+#include "FormationPattern.h"
```
Comment (TomGoodIdea) [inline source/Person.cpp:19]: I'm not sure, but I think you don't need to include this in this case.
```suggestion
```

Comment (petervdmeer) [reply source/Person.cpp:19]: It complains about an incomplete type on line 39 if I leave this include out. (The include is apparently needed for getting the pattern from gamedata.)

Comment (TomGoodIdea) [reply source/Person.cpp:19]: ok

**Review by TomGoodIdea — APPROVED** [review]

**Review by tibetiroka — COMMENTED** [review]
It looks good, so if you tested it, I'm fine with merging it.

**Comment by petervdmeer** [comment]
> It looks good, so if you tested it, I'm fine with merging it.

I've tested it. It works similarly to formations on NPC fleets; the fleets typically don't take the time to get into formation (this gets fixed in other PRs), but when they do, then the formation does appear as intended.

**Pull request merged**